pkg/cmd/cli/config/store.go

```diff
@@ -0,0 +1,169 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+)
+
+const (
+	OpenShiftConfigPathEnvVar      = "OPENSHIFTCONFIG"
+	OpenShiftConfigFlagName        = "config"
+	OpenShiftConfigFileName        = ".openshiftconfig"
+	OpenShiftConfigHomeDir         = ".config/openshift"
+	OpenShiftConfigHomeFileName    = ".config"
+	OpenShiftConfigHomeDirFileName = OpenShiftConfigHomeDir + "/" + OpenShiftConfigHomeFileName
+
+	KubeConfigPathEnvVar = clientcmd.RecommendedConfigPathEnvVar
+	KubeConfigFileName   = ".kubeconfig"
+	KubeConfigHomeDir    = ".kube"
+
+	fromFlag     = "flag"
+	fromEnvVar   = "envvar"
+	fromLocalDir = "localdir"
+	fromHomeDir  = "homedir"
+
+	fromKube      = "fromkube"
+	fromOpenShift = "fromopenshift"
+)
+
+type ConfigStore struct {
+	Config           *clientcmdapi.Config
+	Path             string
+	providerEngine   string
+	providerLocation string
+}
+
+func (c *ConfigStore) FromFlag() bool {
+	return c.providerLocation == fromFlag
+}
+
+func (c *ConfigStore) FromEnvVar() bool {
+	return c.providerLocation == fromEnvVar
+}
+
+func (c *ConfigStore) FromLocalDir() bool {
+	return c.providerLocation == fromLocalDir
+}
+
+func (c *ConfigStore) FromHomeDir() bool {
+	return c.providerLocation == fromHomeDir
+}
+
+func (c *ConfigStore) FromOpenShift() bool {
+	return c.providerEngine == fromOpenShift
+}
+
+func (c *ConfigStore) FromKube() bool {
+	return c.providerEngine == fromKube
+}
+
+func GetConfigFromDefaultLocations(clientCfg *client.Config, cmd *cobra.Command) (*ConfigStore, error) {
```

Comment:
If this is supposed to follow the same loading order and rules as the kubeconfig load, we definitely need to update upstream to take a []string for the loading order so that we can all agree on it.